src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -422,6 +392,28 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+        /// <summary>
+        /// Retrieves the requested subset of rows from the requested result set. Intended to
+        /// </summary>
+        /// <param name="subsetParams">Parameters for the subset to retrieve</param>
+        /// <returns>The requested subset</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The requested query does not exist</exception>
+        public async Task<ResultSetSubset> InterServiceResultSubset(SubsetParams subsetParams)
```

Comment:
That was the bit of the comment that was left off, lol...

Basically I picked the name "interservice" to indicate that it's meant to be called by another service (in this case the EditDataService). This bypasses the request context and implementation that assumes its being called by the JSON RPC. We also use the same methods from the JSON RPC request handlers, so I'm trying to keep the implementations here DRY.